### PR TITLE
propagate listener errors and transport failure from streaming requests

### PR DESCRIPTION
**Pull request by turbocrime**

### CODE REVIEW

File: packages/transport-dom/src/create.ts [hunk 1]
```diff
@@ -127,15 +127,19 @@ export const createChannelTransport = ({
...
-        trailer: new Headers(result.trailer),
-        message: method.O.fromJson(result.message, jsonOptions),
+        header: new Headers((await success).header),
+        trailer: new Headers((await success).trailer),
+        message: await success.then(({ message }) => {
+          const o = new method.O();
+          Any.fromJson(message, jsonOptions).unpackTo(o);
+          return o;
+        }),
       };
```
Comment (turbocrime) [inline packages/transport-dom/src/create.ts:143]: now using the `Any.fromJson` pattern, previously this was erroneously using the message class's method which had subtly different behavior regarding unknown fields

File: packages/transport-dom/src/create.ts [hunk 2]
```diff
@@ -127,15 +127,19 @@ export const createChannelTransport = ({
...
         stream: false,
-        header: new Headers(result.header),
-        trailer: new Headers(result.trailer),
-        message: method.O.fromJson(result.message, jsonOptions),
+        header: new Headers((await success).header),
+        trailer: new Headers((await success).trailer),
+        message: await success.then(({ message }) => {
+          const o = new method.O();
+          Any.fromJson(message, jsonOptions).unpackTo(o);
+          return o;
```
Comment (jessepinho) [inline packages/transport-dom/src/create.ts:141]: What's this change do?

File: packages/transport-dom/src/create.ts [hunk 3]
```diff
@@ -127,15 +127,19 @@ export const createChannelTransport = ({
       const message = Any.pack(new method.I(input)).toJson(jsonOptions);
       port.postMessage({ requestId, message, header });
 
-      const result = await Promise.race([response, listenerError.promise]);
+      const success = Promise.race([response, listenerError.promise]);
```
Comment (jessepinho) [inline packages/transport-dom/src/create.ts:130]: I feel like `success` sounds like a boolean, but 🤷🏻‍♂️

**Comment by turbocrime** [comment]
oops this was reffing the wrong issue

**Pull request closed without merging**